packages/ipfs-core/src/components/repo/gc.js

```diff
@@ -35,6 +35,7 @@ module.exports = ({ gcLock, pin, refs, repo }) => {
       // Mark all blocks that are being used
       const markedSet = await createMarkedSet({ pin, refs, repo })
       // Get all blocks keys from the blockstore
+      // @ts-ignore - TS is not aware of keysOnly overload
```

Comment:
do we have an issue for this ?

Reply:
There is this thread https://github.com/ipfs/interface-datastore/issues/38 and as I mentioned somewhere here I do intend to trickle down interface defs introduces here to various packages as followup work, which I hope will be a driving force to rethink some of the API choices made.

For this specifically it seems like having a separate query method would be simple and pragmatic solution, that does not require some advanced type magic that I'm trying to reduce as much as possible.